janus_core/geo_opt.py

```diff
@@ -0,0 +1,55 @@
+"""Geometry optimisation."""
+
+from ase import Atoms
+
+try:
+    from ase.filters import FrechetCellFilter as DefaultFilter
+except ImportError:
+    from ase.constraints import ExpCellFilter as DefaultFilter
+
+from ase.optimize import LBFGS
+
+
+#  pylint: disable=dangerous-default-value
+#  Dictionaries are not modified within the function.
+def optimize(
+    atoms: Atoms,
+    fmax: float = 0.1,
+    dyn_kwargs: dict = {},
+    filter_func: callable = DefaultFilter,
```

Comment:
If `None` is an option, as provided below, should be `Optional[callable]`

Probably also doesn't want to be generic `callable`, but a subclass of an ASE filter base type if that's a thing.